src/Network/Request.php

```diff
@@ -1375,6 +1381,42 @@ public function setInput($input)
     }
 
     /**
+     * Update the request with a new request data element.
+     *
+     * Returns an updated request object. This method returns
+     * a *new* request object and does not mutate the request in-place.
+     *
+     * @param string $name The dot separated path to insert $value at.
+     * @param mixed $value The value to insert into the request data.
+     * @return self
+     */
+    public function withData($name, $value)
+    {
+        $copy = clone $this;
+        $copy->data = Hash::insert($copy->data, $name, $value);
+
+        return $copy;
+    }
```

Comment:
It just doesn't make for easy reading of the source code. When you have a method name that is a *verb* then *noun* it reads well, but `withData` is a *preposition* followed by a *noun*. It leaves you wanting to know what the hell it's with, but the with part is in the parameters which for English is the wrong order for reading a preposition.

It does work good with method chaining, because the with could be about what came before the method.

```
   $table->update()->withData('foo',true)->execute();
```

> Update with some data!

I'm just rambling on about nothing. Oh let's argue next about the virtues of camel case variable names!